spot-client/src/spot-tv/ui/components/setup/SyncWithBackend.js

```diff
@@ -77,28 +96,56 @@ export class SyncWithBackend extends React.Component {
      * @returns {void}
      */
     _onChange(value) {
-        if (value.length !== 8) {
+        if (value.length !== EXPECTED_LLPC_LENGTH) {
             return;
         }
 
-        const setLoadingPromise = new Promise(resolve => {
-            this.setState({
-                loading: true
-            }, resolve);
+        this.setState({
+            loading: true
+        }, () => {
+            this._sync(value);
         });
+    }
 
-        setLoadingPromise
-            .then(() => this.props.onAttemptSync(value))
-            .then(
-                this.props.onSuccess,
-                error => {
-                    this.setState({
-                        loading: false
-                    });
-
-                    logger.error('connectSpotTvToBackend failed', { error });
-                    this.props.onSyncError(error);
+    /**
+     * Syncs with backend with the provided pairing code.
+     *
+     * @param {string} value - The pairing code to sync with.
+     * @returns {void}
+     */
+    _sync(value) {
+        this.props.onAttemptSync(value).then(
+            this.props.onSuccess,
+            error => {
+                this.setState({
+                    autoSyncing: false,
+                    loading: false
                 });
+
+                logger.error('connectSpotTvToBackend failed', { error });
+                this.props.onSyncError(error);
+            }
+        );
+    }
+
+    /**
+     * Tries to perform an auto sync if a pairing code is provided on start.
+     *
+     * @returns {void}
+     */
+    _tryAutoSync() {
+        const { llpc } = this.props.startParams || {};
+
+        if (!llpc || llpc.length !== EXPECTED_LLPC_LENGTH) {
```

Comment:
What about validating when the params get put into redux? That way validation occurs as close to time of retrieval as possible.

Reply:
Actually I intentionally stayed away from doing that. As you see I preserve the query params at the start of the app in a generic way, there can be literally any keys with any values, and they are stored in the setup Redux tree. I want the features to use them in their own way: every features use what they need, and others don't need to know about them.

Now, if I want to validate these at store time, I need to put feature logic (I know this term doesn't really exist in Spot, but I guess you understand it) into the common reducer which feels kinda weird.

Or did you have something different in your mind? What do you think?

Reply:
What about not validating at all then? And if you're preserving params for a use case which may not exist yet, why do you store them in setup?

Reply:
Not sure where else to store, without being feature (screen) dependant. I get the not-yet-used argument, but to be honest, it would take more time to persist them on a need basis than to persist them all and let the features (screens) use them as they want.
Also, I'm certain that we'll need more params in the future than these 3 (1 is in this commit, 2 more to come into the PR), especially if we implement further integrations.